geopandas/tests/test_sindex.py

```diff
@@ -834,6 +834,53 @@ def test_nearest_max_distance(
         else:
             assert_array_equal(res, expected[0])
 
+    @pytest.mark.skipif(
```

Comment:
Can you also please add a test that the error is raised when Shapely 2.0 isn't available?  Something like (untested)

```Python

@pytest.mark.skipif(compat.USE_SHAPELY_20, reason="sindex.nearest exclusive parameter is available for Shapely 2.0")
def test_nearest_exclusive_unavailable():
    geoms = mod.points(np.arange(5), np.arange(5))
    df = geopandas.GeoDataFrame(geometry=geoms)

    with pytest.raises(NotImplementedError, match="requires shapely >= 2.0"):
        df.sindex.nearest(geoms, exclusive=True)

```

Reply:
thank you for preparing this. It's tested and included now